### ISSUE TITLE
 Resource exhausted by faster rcnn and ssd restnet

### ISSUE THREAD

**Issue by heizie** [tensorflow/models#7166]
same here. The batch size was already 1 and i've change the **fixed_shape_resizer** as 500x500 (using faster rcnn models) and
```
  session_config = tf.ConfigProto()
  session_config.gpu_options.per_process_gpu_memory_fraction = 0.3
  config = tf.estimator.RunConfig(model_dir=FLAGS.model_dir, session_config=session_config)
```
  
is also set.

But it keep showing (by ssd resnet, same error) :
    
`2019-07-08 18:37:10.194834: W tensorflow/core/framework/op_kernel.cc:1502] OP_REQUIRES failed at cwise_ops_common.cc:70 : Resource exhausted: OOM when allocating tensor with shape[100,51150] and type float on /job:localhost/replica:0/task:0/device:GPU:0 by allocator GPU_0_bfc
`
until it break automatically. 
I can only train the ssd mobilenet. very confusing.

i'm using gtx 1060 6G and RTX 2070. both has same error

_Originally posted by @heizie in https://github.com/tensorflow/models/issues/1993#issuecomment-509300382_

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
What is the top-level directory of the model you are using
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by heizie** [comment]
> What is the top-level directory of the model you are using
/home/lin/tf-obj/models/.....
> Have I written custom code
Using script for transforming xml to tfrecord. the images site of tensorboard shows the labels are correct and readable.( there is a error, the inference's output has wrong label name. I've double check label map and xml, they are coorect) 
> OS Platform and Distribution
ubuntu 18.04
> TensorFlow installed from
tensorflow docker
> TensorFlow version
1.13.1
RTX 2070 8GB & GTX 1060 6GB
> Exact command to reproduce
config file:

model {
  faster_rcnn {
    num_classes: 1
    image_resizer {
      fixed_shape_resizer {
        height: 500
        width: 500
      }
    }
    feature_extractor {
      type: "faster_rcnn_nas"
    }
    first_stage_anchor_generator {
      grid_anchor_generator {
        height_stride: 16
        width_stride: 16
        scales: 0.25
        scales: 0.5
        scales: 1.0
        scales: 2.0
        aspect_ratios: 0.5
        aspect_ratios: 1.0
        aspect_ratios: 2.0
      }
    }
    first_stage_box_predictor_conv_hyperparams {
      op: CONV
      regularizer {
        l2_regularizer {
          weight: 0.0
        }
      }
      initializer {
        truncated_normal_initializer {
          stddev: 0.00999999977648
        }
      }
    }
    first_stage_nms_score_threshold: 0.0
    first_stage_nms_iou_threshold: 0.699999988079
    first_stage_max_proposals: 300
    first_stage_localization_loss_weight: 2.0
    first_stage_objectness_loss_weight: 1.0
    initial_crop_size: 17
    maxpool_kernel_size: 1
    maxpool_stride: 1
    second_stage_box_predictor {
      mask_rcnn_box_predictor {
        fc_hyperparams {
          op: FC
          regularizer {
            l2_regularizer {
              weight: 0.0
            }
          }
          initializer {
            variance_scaling_initializer {
              factor: 1.0
              uniform: true
              mode: FAN_AVG
            }
          }
        }
        use_dropout: false
        dropout_keep_probability: 1.0
      }
    }
    second_stage_post_processing {
      batch_non_max_suppression {
        score_threshold: 0.300000011921
        iou_threshold: 0.600000023842
        max_detections_per_class: 100
        max_total_detections: 100
      }
      score_converter: SOFTMAX
    }
    second_stage_localization_loss_weight: 2.0
    second_stage_classification_loss_weight: 1.0
  }
}
train_config {
  batch_size: 1
  data_augmentation_options {
    random_horizontal_flip {
    }
  }
  optimizer {
    momentum_optimizer {
      learning_rate {
        manual_step_learning_rate {
          initial_learning_rate: 0.000300000014249
          schedule {
            step: 0
            learning_rate: 0.000300000014249
          }
          schedule {
            step: 900000
            learning_rate: 2.99999992421e-05
          }
          schedule {
            step: 1200000
            learning_rate: 3.00000010611e-06
          }
        }
      }
      momentum_optimizer_value: 0.899999976158
    }
    use_moving_average: false
  }
  gradient_clipping_by_norm: 10.0
  fine_tune_checkpoint: "......./model.ckpt"
  from_detection_checkpoint: true
  num_steps: 200000
}
train_input_reader {
  label_map_path: ".....label_map.pbtxt"
  tf_record_input_reader {
    input_path: "....record"
  }
}
eval_config {
  num_examples: 8000
  max_evals: 10
  use_moving_averages: false
}
eval_input_reader {
  label_map_path: ".....label_map.pbtxt"
  shuffle: false
  num_readers: 1
  tf_record_input_reader {
    input_path: ".....record"
  }
}


model_main.py
........same as original.....
def main(unused_argv):
  
  session_config = tf.ConfigProto()
  session_config.gpu_options.per_process_gpu_memory_fraction = 0.3
  config = tf.estimator.RunConfig(model_dir=FLAGS.model_dir, session_config=session_config)
    
  flags.mark_flag_as_required('model_dir')
  flags.mark_flag_as_required('pipeline_config_path')
  config = tf.estimator.RunConfig(model_dir=FLAGS.model_dir)

.......

**Comment by heizie** [comment]
**I upgrade the tensorflow version from 1.13 to 1.14. Problem solved**
pip3 install --user --upgrade tensorflow-gpu

**Comment by MonsterLWR** [comment]
I solved this problem by using sharding datasets described in this page:
https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/using_your_own_dataset.md

**Comment by khany27** [comment]
Hello, I am facing the same issue if I train SSD Inception. I did not face problem with Faster RCNN Inception and Atrous until now but training all SSD models give 'Resource exhausted'.

I have installed Tensorflow-gpu 1.15.0 and batch_size is 1

**Comment by sidharth1805** [comment]
Refer here: https://github.com/tensorflow/models/issues/1993#issuecomment-761742221
